packages/core/src/odata-common/entity-builder.ts

```diff
@@ -49,24 +63,44 @@ export class EntityBuilder<EntityT extends Entity, JsonT> {
    * @param json - Representation of the entity in JSON format.
    * @returns Entity constructed from JSON representation.
    */
-  public fromJson(json: Partial<JsonT>): EntityT {
+  public fromJson(json: FromJsonType<JsonT>): EntityT {
     const entityBuilder = this._entityConstructor.builder();
-    Object.entries(json).forEach(([key, value]) => {
-      if (typeof entityBuilder[key] === 'function') {
-        entityBuilder[key](value);
-      }
-      if (key === '_customFields') {
-        entityBuilder.withCustomFields(value as Record<string, any>);
-      }
-    });
+    const entityConstructor = this._entityConstructor;
+
+    const [entityEntries, customEntries] = partition(
```

Comment:
Instead of `Partial `, `FromJsonType` is used, which is more complicated but should be more precise.
I see the type of `entityEntries` has been changed from `[string, unkonwn][]` to `[string, any][]`.
Could you please share more insights about the benefit of the new type?

Reply:
The `Partial` type was just completely wrong ;)
1. Not recursive:
```ts
const bupaJson = {
  firstName: 'Peter',
  to_Adress: [
    city: 'Berlin'
  ]
}
```

If you try to assign it to a variable with `Partial<BusinessPartner>` this will not work, because `Partial` is not recursive and you have to specify every property of the address, that is not optional.

2. Did not allow `_customFields`
There was an implementation to account for `_customFields`, but according to the type, you weren't able to set those.
I deprecated this behavior and instead allowed to set unknown properties directly.